test/boost/sstable_compaction_test.cc

```diff
@@ -2131,6 +2131,41 @@ SEASTAR_THREAD_TEST_CASE(sstable_scrub_validate_mode_test) {
     });
 }
 
+SEASTAR_THREAD_TEST_CASE(sstable_scrub_validate_mode_test_corrupted_file) {
+    scrub_test_framework test;
+
+    auto schema = test.schema();
+
+    auto muts = tests::generate_random_mutations(
+            test.random_schema(),
+            tests::uncompactible_timestamp_generator(test.seed()),
+            tests::no_expiry_expiry_generator(),
+            std::uniform_int_distribution<size_t>(10, 10)).get();
+
+    test.run(schema, muts, [] (table_for_tests& table, compaction::table_state& ts, std::vector<sstables::shared_sstable> sstables) {
+        BOOST_REQUIRE(sstables.size() == 1);
+        auto sst = sstables.front();
+
+        // Corrupt the data to cause an invalid checksum.
+        auto f = open_file_dma(sstables::test(sst).filename(component_type::Data).native(), open_flags::wo).get();
+        const auto wbuf_align = f.memory_dma_alignment();
+        const auto wbuf_len = f.disk_write_dma_alignment();
+        auto wbuf = seastar::temporary_buffer<char>::aligned(wbuf_align, wbuf_len);
+        std::fill(wbuf.get_write(), wbuf.get_write() + wbuf_len, 0xba);
+        f.dma_write(0, wbuf.get(), wbuf_len).get();
+        f.close().get();
+
+        // No way to really test validation besides observing the log messages.
```

Comment:
So will this test pass even if scrub doesn't report an error?

Reply:
I copied this comment from the other scrub tests. It means that scrub does not report its status so we need to figure it out from its side effects (check if the SSTable has been quarantined or check the logs). Since we check the state of the SSTable, the test should fail.

On a closer look, it looks like this comment has now become stale. It was first introduced in https://github.com/scylladb/scylladb/commit/5c39a2921ea where indeed the scrub function would just return an empty future. However, this changed in https://github.com/scylladb/scylladb/pull/11074 and now it reports the number of validation errors.